revoke/revoke.go

```diff
@@ -60,32 +133,49 @@ func ldapURL(url string) bool {
 //
 //  true, false:  failure to check revocation status causes
 //                  verification to fail
-func revCheck(cert *x509.Certificate) (revoked, ok bool) {
+func (r *Revoke) revCheck(cert *x509.Certificate) (revoked, ok bool) {
+	r.lock.Lock()
+	localCRL := r.localCRL
+	hardFail := r.hardFail
+	r.lock.Unlock()
```

Comment:
so we don't lock in certIsRevokedCRL. lock/unlock makes sure the update happens in atomic transaction, simply placing lock/unlock around read/write is not implementing thread safety. I do suggest you study about ACID principle and multithread programming.